src/parquet/column_reader.h

```diff
@@ -148,47 +148,37 @@ static inline void DefinitionLevelsToBitmap(
     const int16_t* def_levels, int64_t num_def_levels, const int16_t max_definition_level,
     const int16_t max_repetition_level, int64_t* values_read, int64_t* null_count,
     uint8_t* valid_bits, const int64_t valid_bits_offset) {
-  int64_t byte_offset = valid_bits_offset / 8;
-  int64_t bit_offset = valid_bits_offset % 8;
-  uint8_t bitset = valid_bits[byte_offset];
+  ::arrow::internal::BitmapWriter valid_bits_writer(valid_bits, valid_bits_offset,
+                                                    num_def_levels);
 
   // TODO(itaiin): As an interim solution we are splitting the code path here
   // between repeated+flat column reads, and non-repeated+nested reads.
   // Those paths need to be merged in the future
   for (int i = 0; i < num_def_levels; ++i) {
     if (def_levels[i] == max_definition_level) {
-      bitset |= (1 << bit_offset);
+      valid_bits_writer.Set();
     } else if (max_repetition_level > 0) {
       // repetition+flat case
       if (def_levels[i] == (max_definition_level - 1)) {
-        bitset &= ~(1 << bit_offset);
+        valid_bits_writer.NotSet();
         *null_count += 1;
       } else {
         continue;
       }
     } else {
       // non-repeated+nested case
       if (def_levels[i] < max_definition_level) {
-        bitset &= ~(1 << bit_offset);
+        valid_bits_writer.NotSet();
```

Comment:
We should call this `Clear` to be consistent with terminology in arrow/util/bit-util.h